### PR TITLE
Separate adding locations from adding IP addresses

### PR DESCRIPTION
**Pull request by quis**
There are two problems with the journey at the moment:
- there are multiple places to add IP addresses from, and it’s not clear which one you should use (this adds cognitive load)
- adding a new location lets you add 5 IP addresses, but adding to an existing organisation only lets you add one at a time
- there’s a `<select>` control for choosing a location, which we generally try to avoid

This commit simplifies the pages by having one button to add a location, and a button for each location that lets you add IP address

Making this journey simpler makes a bit of room to add some contextual help and hint text that should help people complete these tasks.

I’ve also made the ability to add 5 IP addresses available (not just when you’re adding a location) because it felt useful.

# Before 

image

image

*** 

image

image

# After 

image

image

*** 

image

image

### CODE REVIEW

File: app/controllers/locations_controller.rb [hunk 1]
```diff
@@ -25,6 +29,26 @@ def update
...
 
+  def add_ips
+    @location = Location.find(params[:location_id])
+    @location.add_blank_ips(5)
+  end
+
+  def update_ips
+    @location = Location.find(params[:location_id])
+    length_of_ips_before = @location.ips.length
+    if !present_ips.empty? && @location.update(ips_attributes: present_ips)
```
[the remark this answers is not in the document]
Comment (quis) [reply app/controllers/locations_controller.rb]: Apparently `ActionController::Parameters` have an [`.empty?`](https://api.rubyonrails.org/classes/ActionController/Parameters.html#method-i-empty-3F) method but no `.any?` method. 😞

**Comment by freesteph** [comment]
Tragic news that is

Chris Hill-Scott writes:

> quis commented on this pull request.
>
>
>
>> @@ -25,6 +29,26 @@ def update
>      redirect_to(ips_path, notice: 'RADIUS secret key has been successfully rotated')
>    end
>
> +  def add_ips
> +    @location = Location.find(params[:location_id])
> +    @location.add_blank_ips(5)
> +  end
> +
> +  def update_ips
> +    @location = Location.find(params[:location_id])
> +    length_of_ips_before = @location.ips.length
> +    if !present_ips.empty? && @location.update(ips_attributes: present_ips)
>
> Apparently `ActionController::Parameters` have an [`.empty?`](https://api.rubyonrails.org/classes/ActionController/Parameters.html#method-i-empty-3F) method but no `.any?` method. 😞

File: app/controllers/locations_controller.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (freesteph) [reply app/controllers/locations_controller.rb]: 😢

**Comment by quis** [comment]
Deploying to staging for Steve to sign off

**Pull request merged**